src/main/java/io/opentracing/contrib/global/GlobalTracer.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * The <code>GlobalTracer</code> class is not a {@link Tracer} implementation of itself, but instead a utility-class
+ * to centrally reference the {@link #tracer() singleton instance} of the configured {@link Tracer} implementation.
+ * <p>
+ * The singleton instance can be instantiated in one of two ways:
+ * <ol>
+ * <li>Explicitly, by calling {@link #register(Tracer)} with a configured tracer implementation, or:</li>
+ * <li>Automatically by using the Java <code>ServiceLoader</code> SPI mechanism to load an implementation
+ * from the classpath.</li>
+ * </ol>
+ * <p>
+ * Spans and their contexts created by this <code>GlobalTracer</code> will propagate along with standard
+ * <code>ContextSnapshot</code> propagation.
+ *
+ * @author Sjoerd Talsma
+ * @see Tracer
+ */
+public final class GlobalTracer {
+    private static final Logger LOGGER = Logger.getLogger(GlobalTracer.class.getName());
+
+    /**
+     * Use the standard Java SPI <code>ServiceLoader</code> concept to look for a registered Tracer delegate if no
+     * {@link #register(Tracer) explicit delegate} was provided (including a pre-java6 workaround).
+     */
+    private static final Loader<Tracer> DELEGATES = new Loader<Tracer>(Tracer.class);
+
+    /**
+     * The resolved {@link Tracer} to delegate the global tracing implementation to.<br>
+     * This can be either an {@link #register(Tracer) explicitly registered delegate} or an
+     * {@link #DELEGATES automatically resolved Tracer implementation}.<br>
+     * Management of this reference is the responsibility of the {@link #tracer()} method.
+     */
+    private static final AtomicReference<Tracer> delegate = new AtomicReference<Tracer>();
+
+    /**
+     * Private constructor to prevent instantiation of this utility class.
+     */
+    private GlobalTracer() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * This method allows explicit registration of a configured {@link Tracer} implementation to back the behaviour
+     * of the {@link #activeSpan() active global span} objects.
+     *
+     * @param delegate The delegate tracer to delegate the global tracing implementation to.
+     */
+    public static void register(Tracer delegate) {
+        GlobalTracer.delegate.set(delegate != null ? new GlobalSpanTracer(delegate) : null);
+        LOGGER.log(Level.INFO, "Registered GlobalTracer delegate: {0}.", delegate);
+    }
+
+    /**
+     * This method returns the {@link #register(Tracer) explicitly registered} Tracer implementation,
+     * or attempts to lazily load an available implementation according to the standard Java SPI conventions.
+     * <p>
+     * If no delegate is found, the {@link io.opentracing.NoopTracer NoopTracer} will be returned and no
+     * {@link #activeSpan() globally-active spans} will be created.
+     *
+     * @return The non-<code>null</code> global tracer to use.
+     */
+    public static Tracer tracer() {
+        Tracer instance = delegate.get();
```

Comment:
FWIW, if we want to support multiple simultaneous Tracer impls I would rather do it via a `MultiplexedTracer` (or similar). It would use only one of its multiplexed Tracers for SpanContext/Inject/Extract (i.e., "real tracing") but would delegate all other span calls (tags, logs, etc) to the others as well. The only important use case I see for such multiplexing is metrics and logging libs that want to leverage the OT data, and for those I think this pattern could work well.